modules/potree/src/lib/potree-node-source.ts

```diff
@@ -84,36 +117,86 @@ export class PotreeNodesSource extends DataSource<string, PotreeSourceOptions> {
 
   /**
    * Load octree node content
-   * @param path array of numbers between 0-7 specifying successive octree divisions.
+   * @param nodeName name of a node, string of numbers in range 0..7
    * @return node content geometry or null if the node doesn't exist
    */
-  async loadNodeContent(path: number[]): Promise<Mesh | null> {
+  // eslint-disable-next-line max-statements
+  async loadNodeContent(nodeName: string): Promise<Mesh | null> {
     await this.initPromise;
 
     if (!this.isSupported()) {
       return null;
     }
 
-    const isAvailable = await this.isNodeAvailable(path);
+    const isAvailable = await this.isNodeAvailable(nodeName);
     if (isAvailable) {
-      return load(
-        `${this.baseUrl}/${this.metadata?.octreeDir}/r/r${path.join(
-          ''
-        )}.${this.getContentExtension()}`,
+      const result = await load(
+        `${this.baseUrl}/${this.metadata?.octreeDir}/r/r${nodeName}.${this.getContentExtension()}`,
         LASLoader
       );
+
+      if (result) {
+        let projection;
+        if (this.metadata?.projection) {
```

Comment:
I would recommend defining helper methods for reprojection and bounding box calculations.